### DISCUSSION TITLE
Build from source on macOS / Apple Silicon

### DISCUSSION THREAD

**Discussion by nukopia** [PostgREST/postgrest#3139] [Q&A] [answered]
Hi folks,
Apologies for the dumb question, but could someone give some guidance on how to build PostgREST from source on Apple Silicon. I'm new to the Haskell stack, but I followed the instructions at GHCup to get everything ready. Thanks in advance.
-- Iain

**Comment by nukopia** [comment, marked as answer]
Ah, nevermind. Figured it out from reading `.github/scripts/arm/build.sh`, FWIW:

- I'm using Homebrew installed `libpq` and used `brew info libpq` to recall PATH and compiler flags
- ensure GHC 9.2.8 is installed, then;

```bash
# From `brew info libpq` ...
export PATH="/opt/homebrew/opt/libpq/bin:$PATH"
export LDFLAGS="-L/opt/homebrew/opt/libpq/lib"
export CPPFLAGS="-I/opt/homebrew/opt/libpq/include"
# Build it ...
ghcup set ghc 9.2.8
ghcup set cabal latest
cabal v2-update && cabal v2-build
PGRST_BIN=$(cabal exec which postgrest|tail -1)
cp $PGRST_BIN ~/Scripts  # on my machine
~/Scripts/postgrest -e
```

**Reply by steve-chavez** [reply]
Glad you could figure it out! 

Why not use https://formulae.brew.sh/formula/postgrest though?

**Reply by nukopia** [reply]
Ha ha, Steve, now where's the fun in that? ;-)
TBH, I did come to a similar realisation yesterday. A proper "D'oh!" moment.
